Diary-Sample/Models/UserAdminProfileViewModel.cs

```diff
@@ -0,0 +1,20 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace Diary_Sample.Models
+{
+    public class UserAdminProfileViewModel
+    {
+        public string UserId { get; set; } = string.Empty;
+
+        [Required(ErrorMessage = "ユーザ名は必須です。")]
+        [Display(Name = "ユーザ名")]
+        public string UserName { get; set; } = string.Empty;
+        [Required(AllowEmptyStrings = true)]
+        public string PhoneNumber { get; set; } = string.Empty;
```

Comment:
あまり重要でないですが、電話番号が何でも入力できてしまいます。（英字なども）

Reply:
電話番号の形式チェックを入れました。

Reply:
形式がチェックされるとともに必須チェック（サーバ側のみ）も有効になってしまったようです。
↓ 空で入力すると更新されるが、入力欄にエラーメッセージが表示されます。
![スクリーンショット 2021-04-29 144056](https://user-images.githubusercontent.com/15532048/116507817-ffb0cf80-a8fa-11eb-9c98-8827f9f2d258.png)


モデルの検証は[null許容参照型](https://qiita.com/kojimadev/items/d2ee29e85c4c0859275a#null%E8%A8%B1%E5%AE%B9%E5%8F%82%E7%85%A7%E5%9E%8B%E3%81%A8%E3%81%AF)かどうかによって必須チェックしたり、しなかったりするようです。

[Null 非許容の参照型と [Required] 属性](https://docs.microsoft.com/ja-jp/aspnet/core/mvc/models/validation?view=aspnetcore-5.0#non-nullable-reference-types-and-required-attribute)

String型だと必須チェックが行われるので、String?型に修正お願いします。
```suggestion
        public string? PhoneNumber { get; set; } = string.Empty;
```

Reply:
対応方法まで調べてもらってすみません。
ありがとうございました。
こちらでも調べてみたのですが、全然見つかりませんでした…。

この方法で修正させていただきました。
確かにこの方法だと、ちゃんと必須チェックは行わずに、
形式チェックだけ有効になりますね。